src/AddressRegistry.Api.Extract/Extracts/LinkedAddressExtractBuilder.cs

```diff
@@ -1,170 +1,51 @@
 namespace AddressRegistry.Api.Extract.Extracts
 {
-    using System;
-    using System.Collections.Generic;
+    using Projections.Extract.AddressExtract;
     using System.Linq;
     using Be.Vlaanderen.Basisregisters.Api.Extract;
     using Be.Vlaanderen.Basisregisters.GrAr.Extracts;
-    using Be.Vlaanderen.Basisregisters.GrAr.Legacy;
-    using Be.Vlaanderen.Basisregisters.Shaperon;
-    using Microsoft.EntityFrameworkCore;
-    using Projections.Extract;
-    using Projections.Extract.AddressLinkExtract;
     using Projections.Syndication;
-    using Projections.Syndication.Municipality;
-    using Projections.Syndication.StreetName;
+    using Projections.Syndication.AddressLink;
 
     public class LinkedAddressExtractBuilder
     {
-        private readonly ExtractContext _context;
         private readonly SyndicationContext _syndicationContext;
-        private readonly List<MunicipalityLatestItem> _cachedMunicipalities;
-        private readonly List<StreetNameLatestItem> _cachedStreetNames;
-        private const string BuildingUnitObjectType = "Gebouweenheid";
-        private const string ParcelObjectType = "Perceel";
 
-        public LinkedAddressExtractBuilder(ExtractContext context, SyndicationContext syndicationContext)
+        public LinkedAddressExtractBuilder(SyndicationContext syndicationContext)
         {
-            _context = context;
             _syndicationContext = syndicationContext;
-            _cachedMunicipalities = syndicationContext.MunicipalityLatestItems.AsNoTracking().ToList();
-            _cachedStreetNames = syndicationContext.StreetNameLatestItems.AsNoTracking().ToList();
         }
 
-        public IEnumerable<ExtractFile> CreateLinkedBuildingUnitAddressFiles()
+        public ExtractFile CreateLinkedBuildingUnitAddressFiles()
         {
-            var extractItems = _context
-                .AddressLinkExtract
-                .AsNoTracking()
-                .Where(m => m.Complete)
-                .OrderBy(m => m.PersistentLocalId);
-
-            AddressLink TransformRecord(AddressLinkExtractItem r)
-            {
-                var buildingUnitIds = _syndicationContext.BuildingUnitAddressMatchLatestItems.Where(x => x.AddressId == r.AddressId && !x.IsRemoved).Select(x => x.BuildingUnitPersistentLocalId);
-
-                return CreateAddressLink(buildingUnitIds, r, BuildingUnitObjectType);
-            }
-
-            yield return CreateDbfFile<AddressLinkExtractItem, AddressLinkDbaseRecord>(
+            var extractItems =
+                from extractItem in _syndicationContext.AddressBuildingUnitLinkExtract
```

Comment:
would adding .AsNoTracking() after these 2 dbsets speed it up too?

Reply:
I don't think AsNoTracking is necessary here. We select the `dbaserecord` which is a `byte[]` so it won't be trackable.